### PR TITLE
feat(): move request param management to featureserver

### PR DESCRIPTION
**Pull request by rgwozdz**
## koop-core:
- BREAKING CHANGE: remove query parameter coercion (empty string to undefined, "true"/"false" to bool, stringified JSON to JSON) - this should be handled in output plugins or in Koop instance
- BREAKING CHANGE: remove query and body parameter merging - this should be handled in output plugins or in Koop instance

## featureserver

- BREAKING CHANGE: remove support for "limit" query param as it is not part of the Geoservices API specification
- add coercion of query params (empty string to undefined, "true"/"false" to bool, stringified JSON to JSON)
- merge body and query params into one set of request parameters

## Issues Resolved
https://github.com/koopjs/koop/issues/441

### CODE REVIEW

File: packages/featureserver/src/helpers/normalize-request-params.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  query,
+  body,
+  maxRecordCount = MAX_RECORD_COUNT,
+) {
+  const definedQueryParams = _.chain(query)
+    .pickBy(isNotEmptyString)
+    .mapValues(coerceStrings)
+    .value();
+
+  const { resultRecordCount, ...params } = { ...definedQueryParams, ...body };
```
Comment (velveetachef) [inline packages/featureserver/src/helpers/normalize-request-params.js:14]: Is there concern that body params will overwrite query params?

File: packages/featureserver/src/helpers/validate-inputs.spec.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      process.env.KOOP_LOG_LEVEL = undefined;
+    });
+  });
+
+  describe('Validate geojson metadata', () => {
+    it('should validate geojson metadata and find no error', () => {
+      try {
+        validateInputs({}, { metadata: { maxRecordCount: 1 } });
+      } catch (error) {
+        throw new Error('should not have thrown');
```
Comment (velveetachef) [inline packages/featureserver/src/helpers/validate-inputs.spec.js:84]: can remove try catch for tests that re-throw

**Review by velveetachef — APPROVED** [review]

File: packages/featureserver/src/helpers/validate-inputs.spec.js (hunk 2 above)
Comment (rgwozdz) [reply packages/featureserver/src/helpers/validate-inputs.spec.js:84]: Here it doesn't rethrow;  note that I throw a new error.

File: packages/featureserver/src/helpers/normalize-request-params.js (hunk 1 above)
Comment (rgwozdz) [reply packages/featureserver/src/helpers/normalize-request-params.js:14]: That's okay and consistent with Geoservice specification I believe.

Comment (velveetachef) [reply packages/featureserver/src/helpers/normalize-request-params.js:14]: 👍

**Pull request merged**